### ISSUE TITLE
Make EIPs Website a PWA

### ISSUE THREAD

**Issue by Pandapip1** [ethereum/EIPs#5608]
### Proposed Change

This will allow for offline viewing of EIPs just by visiting the EIPs website. I volunteer to implement this.

**Comment by SamWilsn** [comment]
I'm concerned about caching because I don't understand PWAs. Is that something I should be worried about?

**Comment by Pandapip1** [comment]
> I'm concerned about caching because I don't understand PWAs. Is that something I should be worried about?

You can configure custom caching rules. The one I'd implement is "stale while revalidate": Send the cached version immediately (if available) but update the cache in the background.

**Comment by Pandapip1** [comment]
Basically, you can implement bespoke caching rules.

**Comment by lightclient** [comment]
I don't think this is an important use case and so it doesn't warrant more code that needs to be maintained. Unless I misunderstand the scope and this is only a few lines change. IMO a much bigger value add would just be improving the look of the EIPs website.

**Comment by Pandapip1** [comment]
> Unless I misunderstand the scope and this is only a few lines change. 

It'll be about 30 lines of pretty understandable code (including comments).

**Comment by xinbenlv** [comment]
As much as I like PWA's loading speed improvement, not all search engine deal with PWA very well, so I am on the fence for adopting PWA for EIP site

**Comment by Pandapip1** [comment]
> not all search engines deal with PWA very well

As implemented, it shouldn't impact search engine results. It will just be for browser caching.

**Comment by xinbenlv** [comment]
Some search engines don't index well PWA pages. E.g. here are a few things from one of the search engine. just to show it's nontrivial. https://developers.google.com/search/blog/2016/11/building-indexable-progressive-web-apps

**Comment by Pandapip1** [comment]
> Some search engines don't index well PWA pages. E.g. here are a few things from one of the search engine. just to show it's nontrivial. https://developers.google.com/search/blog/2016/11/building-indexable-progressive-web-apps

Pretty sure that that particular reference applies more to SPAs. This particular service worker is a glorified cache-control header.

**Comment by xinbenlv** [comment]
Sure, if you understand the difference between PWA and SPA, just make sure the static HTTP yields the same or similar full content of text, then I have no concern at all for any PWA.

**Comment by Pandapip1** [comment]
Still an issue.

**Comment by Pandapip1** [comment]
There's a PR open to add this.

**Comment by Pandapip1** [comment]
Waiting on approval.

**Comment by Pandapip1** [comment]
Waiting on approval.

**Comment by Pandapip1** [comment]
Waiting on approval.

**Comment by Pandapip1** [comment]
Waiting on approval.

**Comment by Pandapip1** [comment]
This is a comment to dismiss the stale review bot.

**Comment by Pandapip1** [comment]
This is a comment to dismiss the stale review bot.

**Comment by Pandapip1** [comment]
This is a comment to dismiss the stale review bot.

**Comment by Pandapip1** [comment]
Would still like this. It doesn't add much complexity, and is really nice to have.

**Comment by Pandapip1** [comment]
Dismissing bot.

**Comment by Pandapip1** [comment]
Dismissing bot.

**Comment by Pandapip1** [comment]
Dismissing bot.

**Comment by Pandapip1** [comment]
I'll close this. It'll either happen or it won't.
